srv/wordpress/wp-content/plugins/acore-wp-plugin/src/Hooks/WooCommerce/TransmogItem.php

```diff
@@ -0,0 +1,342 @@
+
+        foreach ($items as $item) {
+            if ($item["acore_item_sku"]) {
+                if ($item["acore_item_sku"] == "transmogification") {
+                    $res = $soap->serverInfo();
+                    if ($res instanceof \Exception) {
+                        throw new \Exception(__('The server seems to be offline, try again later!', 'woocommerce'));
+                    }
+                    return;
+                }
+            }
+        }
+    }
+
+    // DO THE FINAL ACTION
+    public static function payment_complete($order_id) {
+        $WoWSrv = ACoreServices::I();
+        $logs = new \WC_Logger();
+        try {
+            $order = new \WC_Order($order_id);
+            $items = $order->get_items();
+
+            $soap = $WoWSrv->getGuildSoap();
+            
+            foreach ($items as $item) {
+                if (isset($item["acore_item_sku"])) {
+                    if ($item["acore_item_sku"] == "transmogification") {
+                        $itemId = $item["acore_transmog_item"];
+                        $charGuid = $item["acore_char_sel"];
+
+                        if (!$itemId) {
+                            throw new \Exception("Fill in an item to transmog!");
+                        }
+
+                        $ACoreSrv = ACoreServices::I();
+                        $charRepo = $ACoreSrv->getCharactersRepo();
+                        $char = $charRepo->findOneByGuid($charGuid);
+
+                        if ($char) {
+                            $data = array('item' => $itemId, 'cname' => $char->getName());
+                            ToolsApi::ItemTransmog($data);
+                          } else {
+                              throw new \Exception("Select a character!");
+                          }
+                    }
+                }
+            }
+        } catch (\Exception $e) {
+            $logs->add("acore_log", $e->getMessage());
+        }
+    }
+
+    public static function showTransmogItem() {
+        ?>
+        <label for="acore_transmog_item">Please enter the item to transmog:</label>
+        <input type="text" maxlength="24" id="acore_transmog_item" class="acore_transmog_item" name="acore_transmog_item" style="width: 300px;">
+        <br><br>
+        <div id="loader-icon">Loading...</div>
+        <div id="item-list-no-content" class="alert alert-info hidden" role="alert">
+            <span>That item doesn't excist</span>
+        </div>
+        <div class="table-responsive hidden" id="itemContainer" style="overflow: auto; height: 300px;">
+            <table class="table table-bordered table-hover align-middle">
+                <tbody style="background: #2c3338;" id="itemList">
+                    <tr class="loading-item-list hidden">
+                    </tr>
+                </tbody>
+            </table>
+        </div>
+        <br>
+
+        <!-- 3D view of character here, not supported yet -->
+
+        <script>
```

Comment:
we do not need to show the transmogrificable items that the players already have, we want to sell NEW items that are good for transmogrification.
For "selling new items" I mean only the aesthetic of the item applied via transmogrification

Reply:
this script can be useful when we'll do this https://github.com/azerothcore/acore-cms/issues/108